Compile error with MacOS 12.3 SDK

After upgrading from MacOS 12.1 SDK to version 12.3, I have experienced the following compile error:

```
[  2%] Building C object trezor-crypto/CMakeFiles/TrezorCrypto.dir/crypto/sha3.c.o
/Users/adam/tw/wallet-core/trezor-crypto/crypto/sha3.c:292:7: error: performing pointer subtraction with a null pointer has undefined behavior [-Werror,-Wnull-pointer-subtraction]
                if (IS_ALIGNED_64(msg)) {
                    ^~~~~~~~~~~~~~~~~~
/Users/adam/tw/wallet-core/trezor-crypto/crypto/sha3.c:29:55: note: expanded from macro 'IS_ALIGNED_64'
#define IS_ALIGNED_64(p) (0 == (7 & ((const char*)(p) - (const char*)0)))
```

--- a/trezor-crypto/crypto/sha3.c
+++ b/trezor-crypto/crypto/sha3.c
@@ -26,7 +26,7 @@
 #define I64(x) x##LL
 #define ROTL64(qword, n) ((qword) << (n) ^ ((qword) >> (64 - (n))))
 #define le2me_64(x) (x)
-#define IS_ALIGNED_64(p) (0 == (7 & ((const char*)(p) - (const char*)0)))
+#define IS_ALIGNED_64(p) (0 == (7 & ((long)(p))))  // [wallet-core] pointer/numerical type, for MacOS SDK 12.3
 # define me64_to_le_str(to, from, length) memcpy((to), (from), (length))
 
 /* constants */

